org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -463,6 +464,15 @@ public void projectConfigurationUpdate(TextDocumentIdentifier param) {
 		return handler.resolveClasspaths(param);
 	}
 
+	/* (non-Javadoc)
+	 * @see org.eclipse.jdt.ls.core.internal.JavaProtocolExtensions#buildWorkspace(java.lang.String)
+	 */
+	@Override
+	public CompletableFuture<BuildWorkspaceResult> buildWorkspace(String type) {
```

Comment:
Suggest buildCodeRunnerWorkspace. Here's why. We don't want to confuse the Eclispe/RedHat guys that we are doing scaffolding works in general. What we are trying to do is to provide a simple code runner scenario.